buildSrc/src/main/kotlin/spha-kotlin-conventions.gradle.kts

```diff
@@ -37,7 +37,11 @@ dependencies {
 java {
     // Adds sources and javadoc to build artifacts. This is relevant for publishing
     withSourcesJar()
-    toolchain { languageVersion = JavaLanguageVersion.of(21) }
+    toolchain { languageVersion = JavaLanguageVersion.of(24) }
+}
+
+kotlin {
+    jvmToolchain(21)
```

Comment:
This looks out of sync with "24" above, or @janniclas?

Reply:
good catch :)

Reply:
fixed and also updated GH actions to support Java 24